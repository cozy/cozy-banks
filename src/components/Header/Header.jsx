import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import styles from 'components/Header/Header.styl'
import CozyTheme from 'cozy-ui/transpiled/react/providers/CozyTheme'

class Header extends React.PureComponent {
  render() {
    const { children, className, theme, fixed, paper, style } = this.props

    return (
      <CozyTheme variant={theme} ignoreItself={false}>
        <div
          role="header"
          className={cx(
            styles[`HeaderColor_${theme}`],
            {
              [styles['Header--fixed']]: fixed,
              [styles['Header--paper']]: paper
            },
            className
          )}
          style={style}
        >
          {children}
        </div>
      </CozyTheme>
    )
  }
}

Header.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  theme: PropTypes.oneOf(['normal', 'inverted'])
}

Header.defaultProps = {
  theme: 'normal',
  fixed: false,
  paper: true
}

export default Header
