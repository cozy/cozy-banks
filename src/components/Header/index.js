import Header from 'components/Header/Header'

export default Header
