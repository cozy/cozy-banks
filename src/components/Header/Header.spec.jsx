import React from 'react'
import { mount } from 'enzyme'
import Header from './Header'
import { useCozyTheme } from 'cozy-ui/transpiled/react/providers/CozyTheme'

describe('Header', () => {
  const setup = element => mount(element).html()

  it(`should display children`, () => {
    expect(setup(<Header>content</Header>)).toMatchSnapshot()
  })

  it(`should extend className`, () => {
    expect(
      setup(<Header className="noPaddingBottom">content</Header>)
    ).toMatchSnapshot()
  })

  it(`should set position fixed`, () => {
    expect(setup(<Header fixed>content</Header>)).toMatchSnapshot()
  })

  it(`should set theme default`, () => {
    expect(setup(<Header theme="normal">content</Header>)).toMatchSnapshot()
  })

  it(`should set theme inverted`, () => {
    expect(setup(<Header theme="inverted">content</Header>)).toMatchSnapshot()
  })

  it('should set cozy theme', () => {
    const Component = () => {
      const theme = useCozyTheme()
      return <>{theme}</>
    }
    expect(
      mount(
        <Header theme="inverted">
          <Component />
        </Header>
      ).text()
    ).toBe('inverted')
  })
})
