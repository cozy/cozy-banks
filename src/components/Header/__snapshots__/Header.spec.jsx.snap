// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should display children 1`] = `"<div class=\\"CozyTheme--normal\\"><div role=\\"header\\" class=\\"HeaderColor_normal Header--paper\\">content</div></div>"`;

exports[`Header should extend className 1`] = `"<div class=\\"CozyTheme--normal\\"><div role=\\"header\\" class=\\"HeaderColor_normal Header--paper noPaddingBottom\\">content</div></div>"`;

exports[`Header should set position fixed 1`] = `"<div class=\\"CozyTheme--normal\\"><div role=\\"header\\" class=\\"HeaderColor_normal Header--fixed Header--paper\\">content</div></div>"`;

exports[`Header should set theme default 1`] = `"<div class=\\"CozyTheme--normal\\"><div role=\\"header\\" class=\\"HeaderColor_normal Header--paper\\">content</div></div>"`;

exports[`Header should set theme inverted 1`] = `"<div class=\\"CozyTheme--inverted\\"><div role=\\"header\\" class=\\"HeaderColor_inverted Header--paper\\">content</div></div>"`;
